awx/main/tasks/jobs.py

```diff
@@ -546,6 +546,11 @@ def run(self, pk, **kwargs):
             status = res.status
             rc = res.rc
 
+            # We call this to get the current values from the database, in case update_model was called
+            # within the threadpools inside of AWXReceptorJob. We use update_model instead of
+            # refresh_from_db here because it contains retry logic that is resilient to database failures.
+            self.instance = self.update_model(self.instance.pk)
```

Comment:
Ping @sarabrajsingh, as he and I discussed error logs that came from `post_run_hook` due to not doing this.

Summarizing here...

Each thread has its own connection object. When connections are dropped, that object is stale. If it's not hitting one particular corner case, it will correct itself after 1 failed query. So post_run_hook errors, but then code after it starts to work again, so it's not easy to observe unless you check the logs.